### PR TITLE
GLSP-77: Allow WebSocket connections to reconnect after interrupt

### PR DESCRIPTION
**Pull request by ndoschek**
If Websocket connections reconnect after an interrupt, ensure that the last modelState is restored on RequestModelAction if available

Part of https://github.com/eclipse-glsp/glsp/issues/77

---

Related to: https://github.com/eclipse-glsp/glsp-client/pull/269

### CODE REVIEW

File: packages/server/src/common/features/model/request-model-action-handler.ts [hunk 1]
```diff
@@ -49,8 +50,24 @@ export class RequestModelActionHandler implements ActionHandler {
         this.logger.debug('Execute RequestModelAction:', action);
         this.modelState.setAll(action.options ?? {});
 
+        const isReconnecting = ClientOptionsUtil.isReconnecting(action.options);
+
         const progress = this.reportModelLoading('Model loading in progress');
-        await this.sourceModelStorage.loadSourceModel(action);
+
+        if (isReconnecting) {
+            const oldModelRoot = this.modelState.root;
```
Comment (tortmayr) [inline packages/server/src/common/features/model/request-model-action-handler.ts:58]: Please extract the reconnect functionality in a protected submethod so that it's easier for adopters to customize the behavior.

File: packages/server/src/common/features/model/request-model-action-handler.ts [hunk 2]
```diff
@@ -49,8 +50,24 @@ export class RequestModelActionHandler implements ActionHandler {
...
+            if (oldModelRoot) {
+                // use current modelRoot of modelState and submit
+                this.modelState.updateRoot(oldModelRoot);
```
Comment (tortmayr) [inline packages/server/src/common/features/model/request-model-action-handler.ts:61]: Why is this necessary? We are basically resuming where we left of before the reconnect. So the modelState and indexing should already  be up to date.

**Review by tortmayr — CHANGES_REQUESTED** [review]
Looks good to me! I only have a few minor suggestions in line

File: packages/server/src/common/features/model/request-model-action-handler.ts (hunk 2 above)
Comment (ndoschek) [reply packages/server/src/common/features/model/request-model-action-handler.ts:61]: Thanks, that's true, I removed that line again!

**Review by tortmayr — APPROVED** [review]
